### PR TITLE
Enhance filter rules input and style improvements

### PR DESCRIPTION
**Pull request by bruceyyu**
- For `range`, add input fields for precise range specification
- For `date range`, add calendar components for better selection
- For `value set` (previously "one of"), add a status checkbox for (un)selecting all at the top of the table
- Add a variable selector for quickly switching variable to filter
- Beautify the overall style of the filter

### CODE REVIEW

File: packages/graphic-walker/src/fields/filterField/slider.tsx [hunk 1]
```diff
@@ -102,6 +93,7 @@ const Slider: React.FC<SliderProps> = React.memo(function Slider ({
     const [dragging, setDragging] = React.useState<'left' | 'right' | null>(null);
     const trackRef = React.useRef<HTMLDivElement | null>(null);
     const sliceRef = React.useRef<HTMLDivElement | null>(null);
+    const [inputValue, setInputValue] = useState(['', '']);
```
Comment (islxyqwe) [inline packages/graphic-walker/src/fields/filterField/slider.tsx:96]: 为什么有两套Value状态，应该没有必要吧

File: packages/graphic-walker/package.json [hunk 2]
```diff
@@ -38,7 +38,9 @@
     "@heroicons/react": "^2.0.8",
     "@kanaries/react-beautiful-dnd": "0.0.2",
     "@kanaries/web-data-loader": "^0.1.7",
+    "@tailwindcss/forms": "^0.5.4",
     "autoprefixer": "^10.3.5",
+    "date-fns": "^2.30.0",
```
Comment (islxyqwe) [inline packages/graphic-walker/package.json:43]: 考虑Intl.DateTimeFormat

File: packages/graphic-walker/src/fields/filterField/filterEditDialog.tsx [hunk 3]
```diff
@@ -82,12 +102,15 @@ const FilterEditDialog: React.FC = observer(() => {
...
-                <span className="inline-flex items-center rounded-full bg-indigo-100 px-3 py-0.5 text-sm font-medium text-indigo-800">
-                    {uncontrolledField.name}
-                </span>
-                <h3 className="text-base font-semibold py-2 outline-none">{t("form.rule")}</h3>
+            <div className="px-4 py-1">
+                <div className="py-1">{t("form.name")}</div>
+                <DropdownSelect
+                    buttonClassName="w-96"
+                    className="mb-2"
+                    options={allFields.map((d) => ({ label: d.name, value: d.fid }))}
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/filterEditDialog.tsx:110]: all field memo

File: packages/graphic-walker/src/fields/filterField/filterEditDialog.tsx [hunk 4]
```diff
@@ -82,12 +102,15 @@ const FilterEditDialog: React.FC = observer(() => {
...
+                    selectedKey={uncontrolledField.fid}
+                    onSelect={(fieldKey) => handleSelectFilterField(fieldKey)}
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/filterEditDialog.tsx:112]: callback

Comment (islxyqwe) [inline packages/graphic-walker/src/fields/filterField/filterEditDialog.tsx:112]: onSelect={handleSelectFilterField}就可以了

File: packages/graphic-walker/src/fields/filterField/slider.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 import { filter, fromEvent, map, throttleTime } from 'rxjs';
+import { useTranslation } from 'react-i18next';
 
 const SliderContainer = styled.div({
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/slider.tsx:6]:
```ts
styled.div`
.color{
}
`
```

File: packages/graphic-walker/src/fields/filterField/slider.tsx [hunk 6]
```diff
@@ -222,12 +213,56 @@ const Slider: React.FC<SliderProps> = React.memo(function Slider ({
...
                     />
                 </SliderTrack>
             </SliderElement>
+            <div className="output">
+                <output htmlFor="slider:min">
+                    <div className="my-1">{t('filters.range.start_value')}</div>
+                    {
+                        <input
+                            type="text"
+                            className="block w-full rounded-md border-0 py-1 px-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-1 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/slider.tsx:228]: mark

File: packages/graphic-walker/src/fields/filterField/tabs.tsx [hunk 7]
```diff
@@ -292,19 +326,72 @@ export const FilterTemporalRangeRule: React.FC<RuleFormProps & { active: boolean
...
+                    <div className="my-1">{t('range.start_value')}</div>
+                    <input
+                        className="block w-full rounded-md border-0 py-1 px-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-1 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
+                        type="datetime-local" 
+                        id="birthdaytime" 
+                        name="birthdaytime"
+                        min={dateStringFormatter(min)}
+                        max={dateStringFormatter(max)}
+                        defaultValue={field.rule.value[0] ? dateStringFormatter(field.rule.value[0]) : dateStringFormatter(min)}
+                        onChange={(e) => {
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/tabs.tsx:348]: mark

File: packages/graphic-walker/src/fields/filterField/tabs.tsx [hunk 8]
```diff
@@ -366,19 +457,42 @@ const Tabs: React.FC<TabsProps> = observer(({ field, onChange, tabs }) => {
     const { t } = useTranslation('translation', { keyPrefix: 'constant.filter_type' });
 
     const [which, setWhich] = React.useState(field.rule?.type ?? tabs[0]!);
-
+    React.useEffect(() => {
```
Comment (ObservedObserver) [inline packages/graphic-walker/src/fields/filterField/tabs.tsx:460]: mark

File: yarn.lock [hunk 9]
```diff
@@ -553,6 +560,77 @@
...
+    react-i18next "^11.18.6"
+    react-shadow "^20.0.0"
+    rxjs "^7.3.0"
+    tailwindcss "^3.2.4"
+    uuid "^8.3.2"
+    vega "^5.22.1"
+    vega-embed "^6.21.0"
+    vega-lite "^5.6.0"
+
+"@kanaries/react-beautiful-dnd@0.0.1":
```
Comment (ObservedObserver) [inline yarn.lock:621]: dnd version

**Review by ObservedObserver — CHANGES_REQUESTED** [review]

File: packages/graphic-walker/package.json (hunk 2 above)
Comment (bruceyyu) [reply packages/graphic-walker/package.json]: removed

File: packages/graphic-walker/src/fields/filterField/slider.tsx (hunk 1 above)
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/slider.tsx]: i've encapsulated input to a separate component

File: yarn.lock (hunk 9 above)
Comment (bruceyyu) [reply yarn.lock]: fixed here e2bd4465c7747f5d14c006637636c02161b6144b

File: packages/graphic-walker/src/fields/filterField/tabs.tsx (hunk 7 above)
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/tabs.tsx]: i've encapsulated calendar input to a separate component

File: packages/graphic-walker/src/fields/filterField/slider.tsx (hunk 6 above)
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/slider.tsx]: i've encapsulated input to a separate component

File: packages/graphic-walker/src/fields/filterField/slider.tsx (hunk 5 above)
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/slider.tsx]: > ```ts
> styled.div`
> .color{
> }
> `
> ```

fixed df73320d56263a81e826f4ccb88e83ab19d3b095

File: packages/graphic-walker/src/fields/filterField/tabs.tsx [hunk 10]
```diff
@@ -259,6 +290,9 @@ export const FilterTemporalRangeRule: React.FC<RuleFormProps & { active: boolean
     const { commonStore } = useGlobalStore();
     const { currentDataset: { dataSource } } = commonStore;
 
+    const { t } = useTranslation('translation', { keyPrefix: 'filters' });
```
[the remark this answers is not in the document]
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/tabs.tsx]: fixed df73320d56263a81e826f4ccb88e83ab19d3b095

File: packages/graphic-walker/src/fields/filterField/slider.tsx [hunk 11]
```diff
@@ -164,23 +158,20 @@ const Slider: React.FC<SliderProps> = React.memo(function Slider ({
         }
     }, [dragging, range, onChange, min, max]);
 
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/slider.tsx]: fixed 9c3ea704c8717098d3337e07ae794430396abe50

File: packages/graphic-walker/src/fields/filterField/tabs.tsx [hunk 12]
```diff
@@ -292,19 +326,72 @@ export const FilterTemporalRangeRule: React.FC<RuleFormProps & { active: boolean
         });
     }, []);
 
+    const dateStringFormatter = (timestamp: number) => {
+        return format(new Date(timestamp), "yyyy-MM-dd'T'HH:mm:ss")
```
[the remark this answers is not in the document]
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/tabs.tsx]: dependency removed and changed to .toISOString

[the remark this answers is not in the document]
Comment (bruceyyu) [reply packages/graphic-walker/src/fields/filterField/tabs.tsx]: > use dayjs

dependency removed and changed to .toISOString

**Pull request merged**